lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -36,13 +36,16 @@ define((require, exports, module) ->
 
     popState = (currentState, stack) ->
       if not stack?
-        return 'start'
+        throw new Error("papeeria highlight rules stack error: stack doesn't exist")
+      if not Array.isArray(stack)
+        return "start"
 
-      if currentState? and currentState != stack[stack.length - 1]
+      if not (currentState == "start" and stack.length == 0) && currentState != stack[stack.length - 1]
```

Comment:
if `stack.length == 0` and `currentState != "start"` then condition to the left of `&&` is evaluated to `true`, you go to the right of `&&` and access stack[-1] which is okay in JS but isnot okay in other languages.

Rewrite it in a simple way:
```
if stack.length == 0
  if currentState != "start" then throw new Error else return "start"

# here we know stack is not empty
if currentState != stack[stack.length-1] then throw new Error
stack.pop()
...
```